enhancements/sig-policy/59-ansible-everyevent-mode/README.md

```diff
@@ -0,0 +1,219 @@
+# Add an "everyEvent" Ansible Job Run Mode to the Policy addon
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+The Open Cluster Management (OCM) Policy addon Ansible integration should support a new run mode of `everyEvent` so that
+an Ansible job can be run for every unique policy violation per managed cluster. This means that if you have two
+noncompliant clusters that a policy applies to, the Ansible job would run only once for every cluster until the cluster
+becomes compliant and then noncompliant again.
+
+## Motivation
+
+The Open Cluster Management (OCM) Policy addon supports running Ansible jobs on policy violations. The Policy addon
+currently supports the run modes `once` and `disabled`. The `once` mode is limiting since after a policy violation
+occurs, the Ansible job runs once and then it is disabled. A user must manually renable it for the Ansible job to run
+again on the next policy violation.
+
+An additional mode of `everyEvent` should be added so that an Ansible job can be run for every unique policy violation
+per managed cluster. A simple example is that a user might use the OCM Policy addon Ansible integration to create
+tickets in their ticketing system on every policy violation per cluster. The new `everyEvent` mode would enable this
+use-case.
+
+### Goals
+
+1. The Policy addon is able to run an Ansible job on every unique policy violation per managed cluster.
+1. Provide an option to avoid excessive Ansible job creation on an unstable policy (e.g. frequest compliant ->
+   noncompliant -> compliant).
+
+### Non-Goals
+
+1. Keeping track of every historical `PolicyAutomation` run.
+1. Adding additional `eventHook` options.
+1. Avoiding infinite creation of Ansible jobs on unstable policies (e.g. frequest compliant -> noncompliant ->
+   compliant).
```

Comment:
@gparvin, I'd like to get your opinion on this.

The situation is: if the PolicyAutomation takes an action on the cluster which resolves the violation, but the policy becomes non-compliant again within the specified `delayAfterRunSeconds`, then the PolicyAutomation will be run again. Potentially in a loop.

As an alternative, I think it could be implemented so that if the policy became non-compliant again during the `delayAfterRunSeconds`, that event would be ignored and not trigger a re-run. It would only be re-run if it transitioned from compliant->noncompliant *after* `delayAfterRunSeconds`.

Like Matt mentioned, if the main use of PolicyAutomation is quick and used to create an alert or a ticket, then re-running like the current proposal is fine. But if the PolicyAutomation is meant to resolve the violation permanently, running a complex playbook each time, then running it on a loop could cause other issues.

I see potential advantages and disadvantages either way, so I just want to be sure it's discussed now.